gateway/src/apicast/policy/routing/request.lua

```diff
@@ -0,0 +1,35 @@
+local setmetatable = setmetatable
```

Comment:
My idea is to use this module from other policies too.

I think we should instantiate it when the policies context is created for the current request, and store an instance of this module there. I think it would improve code organization, and also avoid reading several times the headers and the query arguments.

I'd rather do that in a different PR because it might affect many policies and need changes in some tests.